extensions/amp-apester-media/0.1/amp-apester-media.js

```diff
@@ -291,6 +292,29 @@ class AmpApesterMedia extends AMP.BaseElement {
     return overflow;
   }
 
+  /**
+   * @param {JsonObject} publisher
+   * @return {Element}
+   */
+  constructAmpPixel_(publisher) {
+    if (publisher && publisher['trackingPixel']) {
+      const affiliateId = publisher['trackingPixel']['affiliateId'];
+      const publisherId = publisher['publisherId'];
+      if (affiliateId) {
+        const pixelElement =
```

Comment:
you will need to use the [extension-analytics API](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/amp-components-analytics.md) to do this.

Hope the API will simplify your code here. Let me know if you have any question.